File: employee-service/src/main/java/com/attendance/employeeservice/service/EmployeeService.java
package com.attendance.employeeservice.service;

import com.attendance.common.exception.ResourceNotFoundException;
import com.attendance.employeeservice.entity.Employee;
import com.attendance.employeeservice.repository.EmployeeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EmployeeService {

    @Autowired
    private EmployeeRepository employeeRepository;

    public Employee createEmployee(Employee employee) {
        return employeeRepository.save(employee);
    }

    public Employee getEmployeeById(Long id) {
        return employeeRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Employee not found"));
    }

    public Employee updateEmployee(Long id, Employee employeeDetails) {
        Employee employee = getEmployeeById(id);
        employee.setName(employeeDetails.getName());
        employee.setDepartment(employeeDetails.getDepartment());
        employee.setDesignation(employeeDetails.getDesignation());
        employee.setEmail(employeeDetails.getEmail());
        employee.setStatus(employeeDetails.getStatus());
        employee.setJoinDate(employeeDetails.getJoinDate());
        return employeeRepository.save(employee);
    }

    public void deleteEmployee(Long id) {
        employeeRepository.deleteById(id);
    }

    public List<Employee> getAllEmployees() {
        return employeeRepository.findAll();
    }

    public Page<Employee> searchEmployees(Specification<Employee> spec, Pageable pageable) {
        return employeeRepository.findAll(spec, pageable);
    }
}

File: frontend/src/features/employee/EmployeeService.js
import api from '../../app/api';

const getEmployees = async () => {
  const response = await api.get('/employee/all');
  return response.data.data;
};

const createEmployee = async (employeeData) => {
  const response = await api.post('/employee', employeeData);
  return response.data.data;
};

const updateEmployee = async (employeeData) => {
  const response = await api.put(`/employee/${employeeData.id}`, employeeData);
  return response.data.data;
};

const deleteEmployee = async (employeeId) => {
  const response = await api.delete(`/employee/${employeeId}`);
  return response.data.data;
};

const employeeService = {
  getEmployees,
  createEmployee,
  updateEmployee,
  deleteEmployee,
};

export default employeeService;
